uk.co.compucorp.civicrm.civihrsampledata/civihrsampledata.php

```diff
@@ -130,7 +130,9 @@ function _civihrsampledata_importSampleData($extensionDirectory) {
     $fileToImport = new SplFileObject("{$csvDir}/{$file}.csv");
 
     $importerClassName = "CRM_CiviHRSampleData_Importer_{$class}";
-    $importer = new $importerClassName($fileToImport);
+    $importer = new $importerClassName();
+    $importer->setSplFileObject($fileToImport);
```

Comment:
I see. To be honest I don't a see the problem with calling `parent::__construct($splFileObject)`, but the setter is also good enough as long as you don't forget to update the import() method to check that the $fileHandler is not null